### DISCUSSION TITLE
Unhandled Exception: type '_InternalLinkedHashMap<String, dynamic>' is not a subtype of type 'String'

### DISCUSSION THREAD

**Discussion by scoopy21** [Imperial-lord/mapbox-flutter#12] [Q&A]
@Imperial-lord Can u plz help me?
Although I am using your code I cannot retrieve the 'place' value. Here is the error

> [ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: type '_InternalLinkedHashMap<String, dynamic>' is not a subtype of type 'String'
> E/flutter (14059): #0 getParsedReverseGeocoding (package:testnavigation/helpers/mapbox_handler.dart:42:15)
> E/flutter (14059):
> E/flutter (14059): #1 _LocationFieldState._useCurrentLocationButtonHandler (package:testnavigation/widgets/location_field.dart:63:22)
> E/flutter (14059):

the mapbox_handler code

```
// ----------------------------- Mapbox Reverse Geocoding -----------------------------
Future<Map> getParsedReverseGeocoding(LatLng latLng) async {
  var response =
  json.decode(await getReverseGeocodingGivenLatLngUsingMapbox(latLng));
  Map feature = response['features'][0];
  Map revGeocode = {
    'name': feature['text'],
    'address': feature['place_name'].split('${feature['text']}, ')[1],
    'place': feature['place_name'],
    'location': latLng
  };
  return revGeocode;
}
```
the location_field code

```
_useCurrentLocationButtonHandler() async {
    if (!widget.isDestination) {
      LatLng currentLocation = getCurrentLatLngFromSharedPrefs();

      // Get the response of reverse geocoding and do 2 things:
      // 1. Store encoded response in shared preferences
      // 2. Set the text editing controller to the address
      var response = await getParsedReverseGeocoding(currentLocation);
      sharedPreferences.setString('source', json.encode(response));
      String place = response['place'];
      widget.textEditingController.text = place;
    }
  }
```
Please help

**Comment by Imperial-lord** [comment]
Can you see what is inside the `response['features'][0]` - try printing it once. This looks like a type mismatch issue.
